src/main/java/apoc/trigger/Trigger.java

```diff
@@ -148,80 +149,107 @@ public TriggerInfo toTriggerInfo(Map.Entry<String, Object> e) {
     @Procedure(mode = Mode.WRITE)
     @Description("list all installed triggers")
     public Stream<TriggerInfo> list() {
-        return TriggerHandler.list().entrySet().stream()
+        return TriggerHandler.getInstance().list().entrySet().stream()
                 .map( (e) -> new TriggerInfo(e.getKey(),(String)e.getValue().get("kernelTransaction"),(Map<String,Object>)e.getValue().get("selector"), (Map<String, Object>) e.getValue().get("params"),true, (Boolean) e.getValue().get("paused")));
     }
 
     @Procedure(mode = Mode.WRITE)
     @Description("CALL apoc.trigger.pause(name) | it pauses the trigger")
     public Stream<TriggerInfo> pause(@Name("name")String name) {
-        Map<String, Object> paused = TriggerHandler.paused(name);
+        Map<String, Object> paused = TriggerHandler.getInstance().paused(name);
 
         return Stream.of(new TriggerInfo(name,(String)paused.get("kernelTransaction"), (Map<String,Object>) paused.get("selector"), (Map<String,Object>) paused.get("params"),true, true));
     }
 
     @Procedure(mode = Mode.WRITE)
     @Description("CALL apoc.trigger.resume(name) | it resumes the paused trigger")
     public Stream<TriggerInfo> resume(@Name("name")String name) {
-        Map<String, Object> resume = TriggerHandler.resume(name);
+        Map<String, Object> resume = TriggerHandler.getInstance().resume(name);
 
         return Stream.of(new TriggerInfo(name,(String)resume.get("kernelTransaction"), (Map<String,Object>) resume.get("selector"), (Map<String,Object>) resume.get("params"),true, false));
     }
 
     public static class TriggerHandler implements TransactionEventHandler {
         public static final String APOC_TRIGGER = "apoc.trigger";
-        static ConcurrentHashMap<String,Map<String,Object>> triggers = new ConcurrentHashMap(map("",map()));
-        private static GraphProperties properties;
+        private final ConcurrentHashMap<String,Map<String,Object>> triggers = new ConcurrentHashMap(map("",map()));
+        private final GraphProperties properties;
         private final Log log;
 
         public static final String NOT_ENABLED_ERROR = "Triggers have not been enabled." +
                 " Set 'apoc.trigger.enabled=true' in your neo4j.conf file located in the $NEO4J_HOME/conf/ directory.";
+        private final GraphDatabaseService db;
+        private final AtomicBoolean registeredWithKernel = new AtomicBoolean(false);
 
-        public TriggerHandler(GraphDatabaseAPI api, Log log) {
+        private static TriggerHandler instance;
+
+        private TriggerHandler(GraphDatabaseAPI api, Log log) {
             properties = api.getDependencyResolver().resolveDependency(EmbeddedProxySPI.class).newGraphPropertiesProxy();
 //            Pools.SCHEDULED.submit(() -> updateTriggers(null,null));
             this.log = log;
+            this.db = api;
         }
 
-        public static void checkEnabled() {
+        public static TriggerHandler initialize(GraphDatabaseAPI api, Log log) {
```

Comment:
Should this be `synchronized`, in case multiple callers invoke procedures simultaneously?

Reply:
Oop, nvm, only called from start()